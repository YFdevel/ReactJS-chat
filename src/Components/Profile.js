import React from "react";
import {shallowEqual, useDispatch, useSelector} from "react-redux";
import {changeAge, changeLastName, changeName, showChangeDataMessage, showFieldName} from "../store/actions/profile";
import {createStyles, Input, makeStyles} from "@material-ui/core";
import Button from "@material-ui/core/Button";
import {getProfile} from "../store/selectors/profile";
import classNames from "classnames";
import {useRouteMatch} from "react-router-dom";

const useStyles = makeStyles((theme) => createStyles({
    root: {
        borderColor: theme.palette.primary.main,
        background: "#ffffff"
    },
    inline: {
        display: 'inline',
    },
}));

function Profile() {
    const {showName, name, lastName, age, changedDataMessage} = useSelector(getProfile,shallowEqual);
    const dispatch = useDispatch();
    const classes = useStyles();


    const setShowName = () => {
        dispatch(showFieldName());
    }

    const handleChangeData = (event) => {
        event.preventDefault()
        dispatch(showChangeDataMessage())
        event.target.reset()
        setTimeout(() => {
            dispatch(showChangeDataMessage())
        }, 1500)
    }

    const handleChangeName = (e) => {
        dispatch(changeName(e.target.value))
    }

    const handleChangeLastName = (e) => {
        dispatch(changeLastName(e.target.value))
    }

    const handleChangeAge = (e) => {
        dispatch(changeAge(e.target.value))
    }

    return (

        <div className="profile">

            <h2>Профиль</h2>
            <input className=".input"
                   type="checkbox" onChange={setShowName}
            />

            <span className="profile-user-title-data">Ваши данные</span>

            {showName &&
            <div className="profile-user-data">
                <p>Имя: {name}</p>
                <p>Фамилия: {lastName}</p>
                <p>Возраст: {age}</p>
            </div>
            }
            {changedDataMessage && <div className="changed-data-message">Ваши данные успешно изменены</div>}

            <div className="profile-form-block">
                <form className="form"
                      onSubmit={handleChangeData}
                >

                    <Input
                        type="text"
                        className={classNames(classes.root, "my-Input")}
                        placeholder="Введите ваше имя"
                        required={true}
                        onChange={handleChangeName}
                    />
                    <Input
                        type="text"
                        className={classNames(classes.root, "my-Input")}
                        placeholder="Введите вашу фамилию"
                        required={true}
                        onChange={handleChangeLastName}
                    />
                    <Input
                        type="number"
                        className={classNames(classes.root, "my-Input")}
                        placeholder="Введите ваш возраст"
                        required={true}
                        onChange={handleChangeAge}
                    />
                    <Button
                        type="submit"
                        className={classNames(classes.inline, "my-Button-profile")}
                        variant="contained"
                        color="primary"

                    >
                        Отправить
                    </Button>

                </form>
            </div>

        </div>

    );
}

export default Profile;
