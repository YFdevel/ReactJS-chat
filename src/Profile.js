import React from "react";
import {useDispatch, useSelector} from "react-redux";
import {changeAge, changeLastName, changeName, showChangeDataMessage, showFieldName} from "./store/actions/profile";
import {createStyles, Input, makeStyles} from "@material-ui/core";
import Button from "@material-ui/core/Button";

const useStyles = makeStyles((theme) => createStyles({
    root: {
        borderColor: theme.palette.primary.main,
        background: "#ffffff"
    },
    inline: {
        display: 'inline',
    },
}));

function Profile() {
    const {showName, name, lastName, age, changedDataMessage} = useSelector((state) => state.profile);
    const dispatch = useDispatch();
    const classes = useStyles();


    const setShowName = () => {
        dispatch(showFieldName());
    }

    const handleChangeData = (event) => {
        event.preventDefault()
        dispatch(showChangeDataMessage())
        event.target.reset()
        setTimeout(() => {
            dispatch(showChangeDataMessage())
        }, 1500)
    }

    const handleChangeName = (e) => {
        dispatch(changeName(e.target.value))
    }

    const handleChangeLastName = (e) => {
        dispatch(changeLastName(e.target.value))
    }

    const handleChangeAge = (e) => {
        dispatch(changeAge(e.target.value))
    }

    return (

        <div className="profile">

            <h2>Профиль</h2>
            <input className=".input"
                   type="checkbox" onChange={setShowName}
            />

            <span className="profile-user-title-data">Ваши данные</span>

            {showName &&
            <div className="profile-user-data">
                <p>Имя: {name}</p>
                <p>Фамилия: {lastName}</p>
                <p>Возраст: {age}</p>
            </div>
            }
            {changedDataMessage && <div className="changed-data-message">Ваши данные успешно изменены</div>}

            <div className="profile-form-block">
                <form className="form"
                      onSubmit={handleChangeData} style={{position: "sticky", bottom: "20px"}}
                >

                    <Input
                        type="text"
                        style={{width: "100%", height: "100%", marginBottom: "20px"}}
                        className={classes.root}
                        placeholder="Введите ваше имя"
                        required={true}
                        onChange={handleChangeName}
                    />
                    <Input
                        type="text"
                        style={{width: "100%", height: "100%", marginBottom: "20px"}}
                        className={classes.root}
                        placeholder="Введите вашу фамилию"
                        required={true}
                        onChange={handleChangeLastName}
                    />
                    <Input
                        type="number"
                        style={{width: "100%", height: "100%", marginBottom: "20px"}}
                        className={classes.root}
                        placeholder="Введите ваш возраст"
                        required={true}
                        onChange={handleChangeAge}
                    />
                    <Button
                        type="submit"
                        style={{margin: '10px 0', minHeight: "50px"}}
                        variant="contained"
                        color="primary"
                        className={classes.inline}
                    >
                        Отправить
                    </Button>

                </form>
            </div>

        </div>

    );
}

export default Profile;
