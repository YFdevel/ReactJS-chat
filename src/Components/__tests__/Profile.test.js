import {render} from '@testing-library/react';
import Profile from "../Profile";
import {shallow} from "enzyme";


describe("Profile tests", () => {

    const myProps = {
        name: "Yuriy",
        age: 42
    }


    it("matches snapshot online", () => {
        const component = render(<Profile/>);
        expect(component).toMatchSnapshot();
    })


    it('has an element', () => {
        const component = shallow(<Profile myProps={myProps} showName={true}/>);
        let node = component.find('.profile-user-name');
        expect(node.length).toEqual(1)

    });

    it('should Profile displays  name when props are passed', () => {
        const component = shallow(<Profile name="Yuriy" showName={true}/>);
        let node = component.find('.profile-user-name');
        expect(node.text()).toContain(myProps.name);

    });
})