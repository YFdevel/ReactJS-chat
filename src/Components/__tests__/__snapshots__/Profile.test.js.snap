// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile tests matches snapshot online 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="profile"
      >
        <h2>
          Профиль
        </h2>
        <input
          class=".input"
          type="checkbox"
        />
        <span
          class="profile-user-title-data"
        >
          Ваши данные
        </span>
        <div
          class="profile-form-block"
        >
          <form
            class="form"
          >
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-rootForm-4 my-Input"
              required=""
            >
              <input
                class="MuiInputBase-input MuiInput-input"
                placeholder="Введите ваше имя"
                required=""
                type="text"
                value=""
              />
            </div>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-rootForm-4 my-Input"
              required=""
            >
              <input
                class="MuiInputBase-input MuiInput-input"
                placeholder="Введите вашу фамилию"
                required=""
                type="text"
                value=""
              />
            </div>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-rootForm-4 my-Input"
              required=""
            >
              <input
                class="MuiInputBase-input MuiInput-input"
                placeholder="Введите ваш возраст"
                required=""
                type="number"
                value=""
              />
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-inline-6 my-Button-profile MuiButton-containedPrimary"
              tabindex="0"
              type="submit"
            >
              <span
                class="MuiButton-label"
              >
                Отправить
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="profile"
    >
      <h2>
        Профиль
      </h2>
      <input
        class=".input"
        type="checkbox"
      />
      <span
        class="profile-user-title-data"
      >
        Ваши данные
      </span>
      <div
        class="profile-form-block"
      >
        <form
          class="form"
        >
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-rootForm-4 my-Input"
            required=""
          >
            <input
              class="MuiInputBase-input MuiInput-input"
              placeholder="Введите ваше имя"
              required=""
              type="text"
              value=""
            />
          </div>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-rootForm-4 my-Input"
            required=""
          >
            <input
              class="MuiInputBase-input MuiInput-input"
              placeholder="Введите вашу фамилию"
              required=""
              type="text"
              value=""
            />
          </div>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-rootForm-4 my-Input"
            required=""
          >
            <input
              class="MuiInputBase-input MuiInput-input"
              placeholder="Введите ваш возраст"
              required=""
              type="number"
              value=""
            />
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-inline-6 my-Button-profile MuiButton-containedPrimary"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              Отправить
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
